### PR TITLE
Query: Make translation of Coalesce SqlFunction

### LINKED ISSUE #19227: Query: Relational: Make translation of Coalesce a SqlFunction

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #19227

### CODE REVIEW

File: src/EFCore.Cosmos/Query/Internal/ISqlExpressionFactory.cs [hunk 1]
```diff
@@ -181,15 +181,6 @@ public interface ISqlExpressionFactory
...
             [NotNull] SqlExpression left, [NotNull] SqlExpression right, [CanBeNull] CoreTypeMapping typeMapping = null);
 
-        /// <summary>
-        ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
-        ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
-        ///     any release. You should only use it directly in your code with extreme caution and knowing that
-        ///     doing so can result in application failures when updating to a new Entity Framework Core release.
-        /// </summary>
-        SqlBinaryExpression Coalesce(
-            [NotNull] SqlExpression left, [NotNull] SqlExpression right, [CanBeNull] CoreTypeMapping typeMapping = null);
```
Comment (roji) [inline src/EFCore.Cosmos/Query/Internal/ISqlExpressionFactory.cs]: Ironically Cosmos appears to have a C#-like coalescing operator, which would be better represented as an SqlBinaryExpression. But if we implement translating to that we should probably do it in a separate PR anyway.

/cc @AndriySvyryd

Comment (smitpatel) [reply src/EFCore.Cosmos/Query/Internal/ISqlExpressionFactory.cs]: Already noted here https://github.com/aspnet/EntityFrameworkCore/issues/17722#issuecomment-564312146

Cosmos `??` operator is subtly different from C# `??` operator. In Cosmos, it is about existence of key in given dictionary. So `c["missingKey"] ?? "42"` returns `42` if c does not have missingKey. But if c has the key with null value, it will return null value.

Comment (roji) [reply src/EFCore.Cosmos/Query/Internal/ISqlExpressionFactory.cs]: Good point.
